### PR TITLE
LPS-122232 Update AccountRoleLocalService to scope searches by companyId

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-122232](https://issues.liferay.com/browse/LPS-122232).

Hey @pei-jung, here are my changes. There's a lot here so please let me know if you'd like to talk over it. Please don't forward this as I still have to manually test the upgrade process. However, I'll still invoke CI on this pull.

/cc @pei-jung @alee8888 @ChrisKian @dejuknow @patriciajeanperez

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:stop

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

File: modules/apps/account/account-service/src/main/java/com/liferay/account/internal/upgrade/v1_2_0/UpgradeRole.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+public class UpgradeRole extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		int newType = RoleConstants.TYPE_ACCOUNT;
+		int oldType = RoleConstants.TYPE_PROVIDER;
+
+		String className = StringUtil.quote(
+			"personal", AccountRole.class.getName());
```
Comment (pei-jung) [inline modules/apps/account/account-service/src/main/java/com/liferay/account/internal/upgrade/v1_2_0/UpgradeRole.java:34]: typo? looks like it should be `StringUtil.quote(AccountRole.class.getName(), StringPool.APOSTROPHE)`

**Comment by patriciajeanperez** [comment]
Hi @drewbrokke, the upgrades failures are due to the following error:

`2020-10-16 17:08:43.808 ERROR [main][UpgradeStepRegistratorTracker:144] Failed upgrade process for module com.liferay.account.service
com.liferay.portal.kernel.upgrade.UpgradeException: java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '.role.AccountRolepersonalcom.liferay.account.role.AccountRole) AND Role_.type_ =' at line 1`

Full log [here](https://testray.liferay.com/reports/production/logs/test-1-18/1602864242301/test-portal-acceptance-pullrequest%28master%29/4589/functional-upgrade-tomcat90-mysql57-jdk8_stable/0/0/liferay-log.txt.gz)

File: modules/apps/account/account-service/src/main/java/com/liferay/account/internal/upgrade/v1_2_0/UpgradeRole.java (hunk 1 above)
Comment (pei-jung) [reply modules/apps/account/account-service/src/main/java/com/liferay/account/internal/upgrade/v1_2_0/UpgradeRole.java:34]: This is the cause of the test failure that Patricia mentioned.

Comment (drewbrokke) [reply modules/apps/account/account-service/src/main/java/com/liferay/account/internal/upgrade/v1_2_0/UpgradeRole.java:34]: Thank you! Yes this was a copy-paste error. I'll fix right away and resend.

Comment (drewbrokke) [reply modules/apps/account/account-service/src/main/java/com/liferay/account/internal/upgrade/v1_2_0/UpgradeRole.java:34]: Also I find the result of this mistake really funny:
```syntax to use near '.role.AccountRolepersonalcom.liferay.account.role.AccountRole) AND Role_.type_ =' at```

**Comment by drewbrokke** [comment]
See https://github.com/pei-jung/liferay-portal/pull/716

**Pull request closed without merging**